src/overview/actions.js

```diff
@@ -257,12 +269,70 @@ export const showTags = index => (dispatch, getState) => {
 }
 
 export const filterTag = tag => (dispatch, getState) => {
-    const query = selectors.query(getState())
+    const state = getState()
+    const query = selectors.query(state)
+    const filterTags = selectors.filterTags(state)
+
     const transformedTag = `#${tag.split(' ').join('+')} `
 
-    const newQuery = query.includes(transformedTag)
+    const newQuery = query.includes(transformedTag.slice(0, -1))
         ? query.replace(transformedTag, '') // Either remove it, if already there
         : transformedTag + query // or prepend it, if not there
 
-    dispatch(setQuery(newQuery))
+    dispatch(setQueryTagsDomains(newQuery))
+}
+
+export const setQueryTagsDomains = (input, isEnter) => (dispatch, getState) => {
+    const state = getState()
+    const filterTags = selectors.filterTags(state)
+    const filterDomains = selectors.filterDomains(state)
+    let filterStatus = selectors.showFilter(state)
+    const query = input
+
+    const terms = input.toLowerCase().match(/\S+/g) || []
+
+    terms.forEach(term => {
+        if (input[input.length - 1] === ' ' || isEnter) {
+            if (constants.HASH_TAG_PATTERN.test(term)) {
```

Comment:
`filterStatus` needs to reassigned when suppose a user type `#githubgithub.com` and now user insert `+` in between the `#github` and `github.com` in the address bar and enter then we have to open the filter because we are entering two things in different filters.

Reply:
Right, I see now your `showFilter` action's reducer actually works like a toggle, rather than setting the state to true (the verb naming conventions are important). Regardless, this thunk should not need to know anything about the current filter state - all it needs to do is set the filter state to be shown. So if you have a `showFilter` action that makes sure the state is set to `true`, just dispatch that instead of checking the current state first - or the suggestion from yesterday about just making your reducers for add/del tag/domain filter actions set the `showFilter` state to true (this is the easiest and cleanest option - never need to worry about that state anymore)

Reply:
Yeah, that makes sense. :)